app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyResponse.java

```diff
@@ -77,24 +97,11 @@ public long getSurveyCreatedOn() {
     public void setSurveyCreatedOn(long surveyCreatedOn) {
         this.surveyCreatedOn = surveyCreatedOn;
     }
-    @Override
-    @JsonIgnore
-    @DynamoDBAttribute
-    public String getHealthCode() {
-        return healthCode;
-    }
-    @Override
-    public void setHealthCode(String healthCode) {
-        this.healthCode = healthCode;
-    }
-    @Override
-    @DynamoDBHashKey
-    @JsonIgnore
-    public String getGuid() {
-        return healthCode + ":" + identifier;
-    }
-    public void setGuid(String guid) {
-        // Ignore. Always calculated as above
+    @DynamoDBIgnore
+    public void setSurveyKey(GuidCreatedOnVersionHolder keys) {
+        this.surveyGuid = keys.getGuid();
+        this.surveyCreatedOn = keys.getCreatedOn();
+        this.surveyKey = String.format("%s:%s", keys.getGuid(), Long.toString(keys.getCreatedOn()));
     }
```

Comment:
Survey really shouldn't be in this object, it's only attached to create a compound survey/response object when generating JSON. I've moved this to the server and created an object, SurveyResponseWithSurvey, to represent that compound object. Once that's moved, there's only surveyKey and there's no ambiguity about how it is set (it's set or it's not).